cursive/src/backends/wasm.rs

```diff
@@ -0,0 +1,266 @@
+                        cloned.borrow_mut().push_back(Event::Char(c));
+                    }
+                }            
+            }
+        }) as Box<dyn FnMut(_)>);
+        document.add_event_listener_with_callback("keydown", closure.as_ref().unchecked_ref())
+            .map_err(|_| std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Failed to add event listener",
+            ))?;
+        closure.forget();
+
+        let buffer = vec![TextColorPair::new(' ', color.clone()); 10_000];
+        let width = canvas.width() as usize;
+        let height = canvas.height() as usize;
+
+        let c = Backend {
+            canvas,
+            width,
+            height,
+            color: RefCell::new(color),
+            events,
+            buffer: RefCell::new(buffer),
+        };
+        Ok(Box::new(c))
+    }
+
+    /// Creates a new Cursive root using a wasm backend.
+    pub fn init() -> std::io::Result<Box<dyn backend::Backend>> {
+        let document = web_sys::window()
+            .ok_or(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Failed to get window",
+            ))?
+            .document()
+            .ok_or(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Failed to get document",
+            ))?;
+        let canvas = document.get_element_by_id("cursive-wasm-canvas")
+            .ok_or(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Failed to get window",
+            ))?
+            .dyn_into::<HtmlCanvasElement>()
+            .map_err(|_| std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Failed to cast canvas",
+            ))?;
+        canvas.set_width(1000);
+        canvas.set_height(1000);
+
+        Self::new(canvas)
+    }
+}
+
+impl cursive_core::backend::Backend for Backend {
+    fn poll_event(self: &mut Backend) -> Option<Event> {
+        self.events.borrow_mut().pop_front()
+    }
+
+    fn set_title(self: &mut Backend, title: String) {
+        self.canvas.set_title(&title);
+    }
+
+    fn refresh(self: &mut Backend) {
+        // web_sys::console::time_with_label("refresh");
+        let data = self.buffer.borrow().clone();
+        paint(text_color_pairs_to_bytes(&data));
+        // web_sys::console::time_end_with_label("refresh");
+    }
+
+    fn has_colors(self: &Backend) -> bool {
+        true
+    }
+
+    fn screen_size(self: &Backend) -> Vec2 {
+        Vec2::new(self.width, self.height)
+    }
+
+    fn print_at(self: &Backend, pos: Vec2, text: &str) {
+        let color = (*self.color.borrow()).clone();
+        let mut buffer = self.buffer.borrow_mut();
+        for (i, c) in text.chars().enumerate() {
+            let x = pos.x + i;
+            buffer[self.width * pos.y + x] = TextColorPair::new(c, color.clone());
+        }
+    }
+
+    fn clear(self: &Backend, _color: cursive_core::theme::Color) {
+    }
```

Comment:
Should be rather simple to implement too?
Just just be something like:

```suggestion
    fn clear(self: &Backend, color: cursive_core::theme::Color) {
        self.buffer.borrow_mut().fill(TextColorPair::new(' ', color.clone()));
    }
```
(untested tho)